app/Models/Foundation/Summit/SelectionPlan.php

```diff
@@ -974,4 +985,72 @@ public function setSubmissionLockDownPresentationStatusDate(?DateTime $submissio
     public function clearSubmissionLockDownPresentationStatusDate():void{
         $this->submission_lock_down_presentation_status_date = null;
     }
+
+    /**
+     * @return ArrayCollection
+     */
+    public function getPresentationActionTypes(): ArrayCollection {
+        return $this->presentation_action_types->map(function ($entity) {
+            return $entity->getType();
+        });
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getPresentationActionTypesMaxOrder(): int {
+        $criteria = Criteria::create();
+        $criteria->orderBy(['order' => 'DESC']);
+        $res = $this->presentation_action_types->matching($criteria)->first();
+        return $res === false ? 0 : $res->getOrder();
+    }
+
+    /**
+     * @param PresentationActionType $presentation_action_type
+     */
+    public function addPresentationActionType(PresentationActionType $presentation_action_type) {
+        $order = $this->getPresentationActionTypesMaxOrder();
```

Comment:
u need to check here if the PresentationActionType is already assigned or not using a query DQL

Reply:
done